src/authz.js

```diff
@@ -1,40 +1,64 @@
 /**
  * The authorizer is responsible for taking User objects that only have
  * an identity set, and adding a collection of roles based on that
- * identity.  This is done by means of a set of plugins in the `authz`
- * directory and named in the configuration.
+ * identity.
  */
 class Authorizer {
   constructor(cfg) {
-    this. authorizers = cfg.app.authorizers.map((name) => {
-      return new (require('./authz/' + name))({cfg});
-    });
   }
 
   async setup() {
-    await Promise.all(this.authorizers.map(authz => authz.setup()));
   }
 
   /**
-   * Authorize the given user, calling user.addRole for all roles
-   * linked to user.identity.
+   * Authorize the given user.
+   * This method takes a User and attaches roles to it
    */
   async authorize(user) {
-    /* invoke all authorizers which recognize this identityProviderId */
-    return Promise.all(this.authorizers.map(authz => {
-      if (authz.identityProviders.indexOf(user.identityProviderId) !== -1) {
-        return authz.authorize(user);
+    const email = user.identityId;
+
+    // grant the everybody role to anyone who authenticates
+    user.addRole('everybody');
+
+    const mozGroupPrefix = 'mozilliansorg_';
+    const mozGroups = [];
+    const ldapGroups = [];
+
+    // Non-prefixed groups are what is known as Mozilla LDAP groups.
+    // Groups prefixed by a provider name and underscore are provided
+    // by a specific group engine. For example `providername_groupone`
+    // is provided by `providername`.
+    // Per https://goo.gl/bwWjvE
+    user.groups.forEach(group => {
+      // capture mozillians groups
+      if (group.indexOf(mozGroupPrefix) === 0) {
+        mozGroups.push(group.replace(mozGroupPrefix, ''));
+      } else if (group.indexOf('_') === -1) {
```

Comment:
I don't think this will work -- we have lots of LDAP groups with `_` in them like `team_moco` or `vpn_releng`.  I think this must be a simple `} else {`.  It might be nice to make a configurable list of providers that gets looped over, with anything not matching assumed to be an LDAP group.  Then if we learn that IAM has added new providers, it's an easy config change.